here-naksha-app-service/src/main/java/com/here/naksha/app/service/http/tasks/WriteFeatureApiTask.java

```diff
@@ -137,4 +130,97 @@ protected void init() {}
     // transform WriteResult to Http FeatureCollection response
     return transformWriteResultToXyzCollectionResponse(wrResult, Storage.class);
   }
+
+  private @NotNull XyzResponse executeUpdateFeatures() throws Exception {
```

Comment:
nth: I was wondering if we could extract some of this function's parts to separate private methods?
It's subjective but I am not a fan of comments inside code - for instance 
```
// Parse API parameters
... code ...
```
could be extracted to `parseApiParameters()` private method. The same applies to `validateParameters`.
This is as informative as comment and it makes the "main function" more readable + you can extract some common logic that is also used by other `execute...` methods.

As said before - it is subjective so don't feel pressured :) I just think that composing logic of small functions is easier on the reader's eyes and makes code cleaner

Reply:
thanks, I pulled out some shared private methods for the URI tags, but it's tricky for the other params, as they are strings but validating them might result in sending error XyzResponses, and each execute method is a bit different about which params to target